api/src/plugins/mail-providers/ses.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  SESClient,
+  SESClientConfig,
+  SendEmailCommand
+} from '@aws-sdk/client-ses';
+
+import { MailProvider, SendEmailArgs } from '../mailer';
+import { SES_ID, SES_SECRET } from '../../utils/env';
+
+export class SESProvider implements MailProvider {
+  private client: SESClient;
+
+  constructor() {
+    if (!SES_ID || !SES_SECRET) {
+      throw new Error('Email service is set to SES but missing required keys.');
+    }
```

Comment:
Since we are checking the truthiness of `SES_ID` in `app.ts`, why don't we change the logic to:
- Check that both `SES_ID` and `SES_SECRET` are truthy
- Pass the values to the constructor

This way we aren't having to check they exist multiple times?

Reply:
Good point, this isn't great. However, I think the solution is just to have a `EMAIL_PROVIDER` variable that's either `ses` or `nodemailer` and use that in `app.ts`.

My thinking is just that that encapsulates the providers a bit better and means `SES_ID` goes back to having one job, not two.

What do you reckon?

Reply:
Yeah, it would have been quicker if I'd just copied more and thought less. Lesson learned.